test/Concurrency/sendable_checking.swift

```diff
@@ -51,6 +51,7 @@ func testCV(
 
   acceptCV(ns3) // expected-warning {{conformance of 'NS3' to 'Sendable' is only available in macOS 11.0 or newer}}
   // expected-note @-1 {{add 'if #available' version check}}
+  // expected-note @-2 {{change the @available attribute of the global function on macOS from 10.15 to 11.0}}
```

Comment:
Also, I see you test the fix-it along with the note in some of these files, but you should probably do it in all of them. (If you don't expect a fix-it, you can assert that by writing `{{none}}`—but I'm thinking you probably shouldn't emit this note without a fix-it, so maybe that's behavior you ought to change!)

Reply:
> I think it would be more typical to emit this note at the source location of the `@available` attribute it will change, not on the call that caused the diagnostic.
> 
> (This is true for the other test cases, too.)

At first I also planned to put this note at the source location of the `@available` attribute, but that would introduce a new problem. If there more than one call caused the diagnostic, it would result in multiple waning and notes in the same location; Do you have any suggestions for that?

Reply:
> Also, I see you test the fix-it along with the note in some of these files, but you should probably do it in all of them. (If you don't expect a fix-it, you can assert that by writing `{{none}}`—but I'm thinking you probably shouldn't emit this note without a fix-it, so maybe that's behavior you ought to change!)

Regarding fix-it, in the file *test/Sema/availability_define.swift*, the `-define-availability` is used to declare `available`, which causes crashes during **fix-it** testing. Do you have any suggestions?

```swift
// RUN: %target-typecheck-verify-swift \
// RUN:   -define-availability "_iOS13Aligned:macOS 10.15, iOS 13.0" \

// ...

@available(_iOS13Aligned, *)
func client() {
// ...
}
```

Reply:
> the -define-availability is used to declare available, which causes crashes during fix-it testing

Oh, interesting! I think you'll probably need to debug this issue to figure out why it's happening. Starting points:

* Where in the compiler is the crash? Is it in the `DiagnosticEngine` class? The `DiagnosticVerifier` class? Is the function you modified on the call stack?
* What are the source locations you're using from the `AvailableAttr`? Are any of them invalid? (If so, you might be able to add an `if` that checks if they're invalid and skips this diagnostic.)

If you get stuck, push the version of your code where you see the crash and I'll try to debug it.

Reply:
> If there more than one call caused the diagnostic, it would result in multiple waning and notes in the same location; Do you have any suggestions for that?

I don't have any suggestion for this, but not sure it would be a big problem and it would definitely help to have the note located at the declaration such as users will have better understanding of the availability attribute looking into declaration. Multiple notes, users will be able to reason about them as well IMO.